src/main/webapp/get-blur-areas.js

```diff
@@ -0,0 +1,235 @@
+
+              // throw error in order not to go to the next .then() statement
+              throw new Error(ERROR_MESSAGE);
+            }
+
+            return response.json();
+          })().then(
+              (jsonBlurAreas) => {
+                if (!Array.isArray(jsonBlurAreas)) {
+                  reject(new Error('Broken json'));
+                  return;
+                }
+
+                // create image to check that all the rectangles
+                // are no greater than the image
+                const imageObject = new Image();
+                const imageUrl = URL.createObjectURL(image);
+
+                imageObject.onload = function() {
+                  const blurAreas = [];
+
+                  for (const jsonRect of jsonBlurAreas) {
+                    let rect;
+                    try {
+                      rect = new Rect(jsonRect, imageObject);
+                    } catch (error) {
+                      // rectangular is invalid - skip it
+                      continue;
+                    }
+
+                    blurAreas.push(rect);
+                  }
+
+                  if (blurAreas.length === 0) {
+                    alert('We did not recognize any areas to blur :(');
+                  }
+
+                  resolve(blurAreas);
+                };
+
+                imageObject.src = imageUrl;
+              },
+
+              (error) => {
+                if (error.message !== response.status + ' server error') {
+                  alert('Unknown error while parsing results.' +
+                      '\nPlease reload the page and try again ' +
+                      'after some time.');
+                  reject(error);
+                }
+              },
+          );
+        },
+
+        (error) => {
+          alert('Unknown error while uploading image.' +
+              '\nPlease reload the page and try again after some time.');
+          reject(error);
+        },
+    );
+  });
+}
+
+/**
+ * Constructor for rectangle to blur.
+ * Takes as parameter rectangle from server - array with 4 points,
+ * point is an object with properties 'x' and 'y'.
+ * Validates rectangle from server.
+ * @param {Array<Object>} rect
+ * @param {Image} image
+ * @constructor
+ */
+function Rect(rect, image) {
+  if (!Array.isArray(rect)) {
+    throw new Error('Not a rectangle');
+  }
+
+  // has 4 points
+  if (rect.length !== 4) {
+    throw new Error('Not a rectangle');
+  }
+
+  // points must have x and y properties
+  for (const point of rect) {
+    if (!point.hasOwnProperty('x') || !point.hasOwnProperty('y')) {
+      throw new Error('Not a rectangle');
+    }
+  }
+
+  // points must not duplicate
+  for (let i = 0; i < rect.length; i++) {
+    for (let j = 0; j < rect.length; j++) {
+      if (i === j) {
+        continue;
+      }
+
+      if (rect[i].x === rect[j].x && rect[i].y === rect[j].y) {
+        throw new Error('Not a rectangle');
```

Comment:
You could be more specific. Something like 'Duplicate points ' + i + ' and ' + j + '(' + rect[i].x + ', ' + rect[i].y + ')'